### DISCUSSION TITLE
Noby's Feb 9 visit to JHU

### DISCUSSION THREAD

**Discussion by nobyhata** [SmartNeedle/SystemIntegration#4] [General]
Hi Dimitri, Lisa, and Marianna

Time: 9am, Wednesday Feb 9
Place: JHU CISST and Zoom (https://partners.zoom.us/my/snrlab)

Experiment A&B I Each party (three of you) start "Launching whole System” “to run simulation”

Experiment A - All virtual
Test I- Condition ) NeedleGuide publish values (x0=0, z0=0, 0<y[0…100]<100, 0<theta [ 0….180] <180
Success Metric I-1) ros2_needle_shape_publisher receives it, or not, Metric I-2) Trajcontrol receive them, or not

Test II -Condition) In response to Test I condition ros2_needle_shape_publisher publish 100 pre-set values of (“Strain Data”) all showing straight needle shape
Metric II-1) Trajcontrol receive them, or not

Test III - Condition) In response to Test II condition, Trajcontrol send dxi=0, dzi=0 to NeedleGuide, Needle Guiude produce xi=0+dx, zi=0+dzi 
Metric III-1) Needle Guide Issues new parameter

Experiment B - All virtual
Test I- Condition ) NeedleGuide publish values (x0=0, z0=0, 0<y[0…100]<150, 0<theta [ 0….180] <180
Success Metric I-1) ros2_needle_shape_publisher receives it, or not, Metric I-2) Trajcontrol receive them, or not

Test II -Condition) In response to Test I condition ros2_needle_shape_publisher publish 100 pre-set values of (“Strain Data”) all showing straight needle shape
Metric II-1) Trajcontrol receive them, or not

Test III - Condition) In response to Test II condition, Trajcontrol send dxi=random, dzi=random to NeedleGuide, Needle Guide produce xi=random+dx, zi=random+dzi 
Metric III-1) Needle Guide Issues new parameter

Test C I Hopkins computer run Shape-sensing needle with sensored needle (physical), and needle guide virtual

Experiment C - Sensoreized needle Physical, Needle Guide Virtual
Test I- Condition ) NeedleGuide publish values (x0=0, z0=0, 0<y[0…100]<150, 0<theta [ 0….180] <180
Success Metric I-1) ros2_needle_shape_publisher receives it, or not, Metric I-2) Trajcontrol receive them, or not

Experiment II -Condition) In response to Test I condition ros2_needle_shape_publisher publish “strain"
Metric II-1) Trajcontrol receive them, or not

**Comment by dlezcan1** [comment]
For test II:

First, calibrate FBG sensors by:
```
ros2 run hyperion_interrogator calibrate_sensors --ros-args -r __ns:=/needle
```


To find the strain values, subscribe to topic: `/needle/state/curvatures` by

```
ros2 topic echo /needle/state/curvatures
```

**Comment by tokjun** [comment]
I have updated the system diagram on GitHub. 

https://github.com/SmartNeedle/SystemIntegration/blob/main/Documents/index.md

This is a diagram we discussed last summer and I assume there have been many changes since then. Please review the diagram and feel free to update. The source code of the diagram can be found at:

https://github.com/SmartNeedle/SystemIntegration/blob/main/Documents/system_diagram.txt

**Reply by tokjun** [reply]
When you review the diagram, please make sure that the names are consistent. We should use this diagram to define the names of the components and use them consistently. We should also add actual ROS node names in the diagram.

**Reply by dlezcan1** [reply]
I have reviewed and updated my part of it. Changes can be found in commit statements. I have not added ROS node names yet, however.

**Reply by tokjun** [reply]
Thanks @dlezcan1 !
